### ISSUE TITLE
Application crash

### ISSUE THREAD

**Issue by SirMorfield** [niklasf/fishnet#94]
Hello devs, there seems to be an problem, the program crashes after some time rendering.
System info:
Debian 9 65bit
Server fully stable
5 of 8 cores used.

Error beneath:

```
><> 1: DEBUG: No job found. Backing off 0.5s
><> 1: DEBUG: No job found. Backing off 1.5s
><> 1: DEBUG: No job found. Backing off 2.6s
><> 1: DEBUG: No job found. Backing off 2.6s
><> 1: DEBUG: No job found. Backing off 3.8s
><> 1: DEBUG: No job found. Backing off 4.8s
><> 1: DEBUG: No job found. Backing off 5.7s
><> 1: DEBUG: No job found. Backing off 6.3s
><> 1: DEBUG: No job found. Backing off 6.6s
><> 1: DEBUG: No job found. Backing off 5.6s
><> 1: DEBUG: No job found. Backing off 10.0s
><> 1: ERROR: Fatal error in worker
Traceback (most recent call last):
  File "/home/user1/.local/lib/python2.7/site-packages/fishnet.py", line 626, in run
    self.run_inner()
  File "/home/user1/.local/lib/python2.7/site-packages/fishnet.py", line 661, in run_inner
    timeout=HTTP_TIMEOUT)
  File "/home/user1/.local/lib/python2.7/site-packages/requests/sessions.py", line 581, in post
    return self.request('POST', url, data=data, json=json, **kwargs)
  File "/home/user1/.local/lib/python2.7/site-packages/requests/sessions.py", line 533, in request
    resp = self.send(prep, **send_kwargs)
  File "/home/user1/.local/lib/python2.7/site-packages/requests/sessions.py", line 646, in send
    r = adapter.send(request, **kwargs)
  File "/home/user1/.local/lib/python2.7/site-packages/requests/adapters.py", line 498, in send
    raise ConnectionError(err, request=request)
ConnectionError: ('Connection aborted.', BadStatusLine("''",))
Traceback (most recent call last):
  File "/usr/lib/python2.7/runpy.py", line 174, in _run_module_as_main
    "__main__", fname, loader, pkg_name)
  File "/usr/lib/python2.7/runpy.py", line 72, in _run_code
    exec code in run_globals
  File "/home/user1/.local/lib/python2.7/site-packages/fishnet.py", line 1940, in <module>
    sys.exit(main(sys.argv))
  File "/home/user1/.local/lib/python2.7/site-packages/fishnet.py", line 1924, in main
    sys.exit(commands[args.command](args))
  File "/home/user1/.local/lib/python2.7/site-packages/fishnet.py", line 1577, in cmd_run
    raise worker.fatal_error
pip._vendor.requests.exceptions.ConnectionError: ('Connection aborted.', BadStatusLine("''",))
```

**Comment by niklasf** [comment]
Hi, thanks for reporting. I think I have found the issue, but to confirm, can you please try

```
$ python2
>>> import requests
>>> isinstance(requests.ConnectionError(), requests.RequestException)
True
```

and confirm if the output is `True` or `False`?

**Comment by SirMorfield** [comment]
It returns True

**Comment by niklasf** [comment]
Thanks, that means I did not find the issue yet :/

What's your `requests` version? It's printed in the summary when starting fishnet. For example:

```
### Checking configuration ...

Python:           2.7.15 (with requests 2.21.0)
```

**Comment by SirMorfield** [comment]
It says:
`
### Checking configuration ...

Python:           2.7.15rc1 (with requests <module 'pip._vendor.requests.__version__' from '/home/user1/.local/lib/python2.7/site-packages/requests/__version__.pyc'>)
`
When opening the file with nano:
`
^C�
^X[3\c^@^@^@^@^@^@^@^@^A^@^@^@@^@^@^@s@^@^@^@d^@^@Z^@^@d^A^@Z^A^@d^B^@Z^B^@d^C^@Z^C^@d^D^@Z^D^@d^E^@Z^E^@d^F^@Z^F^@d^G^@Z^G^@d^H^@Z^H^@d        ^@Z     ^@d
^@S(^K^@^@^@t^H^@^@^@requestss^W^@^@^@Python HTTP for Humans.s^Z^@^@^@http://python-requests.orgs^F^@^@^@2.21.0i^@!^B^@s
^@^@^@Kenneth Reitzs^S^@^@^@me@kennethreitz.orgs
^@^@^@Apache 2.0s^\^@^@^@Copyright 2018 Kenneth Reitzu^L^@^@^@✨ 🍰 ✨N(
^@^@^@t ^@^@^@__title__t^O^@^@^@__description__t^G^@^@^@__url__t^K^@^@^@__version__t    ^@^@^@__build__t
^@^@^@__author__t^P^@^@^@__author_email__t^K^@^@^@__license__t
^@^@^@__copyright__t^H^@^@^@__cake__(^@^@^@^@(^@^@^@^@(^@^@^@^@s6^@^@^@/tmp/pip-build-4hmlUc/requests/requests/__version__.pyt^H^@^@^@<module>^E^@^@^@s^R^@^@^@^F^A^F^A^F^A^F^A^F^A^F^A^F^A^F^A^F^A
`
Any thoughts?

**Comment by niklasf** [comment]
Ok, that might have helped a bit. Meanwhile I have set up a VM with Debian Stretch to try and reproduce the exact circumstances.

Do you remember how exactly you installed Python, pip and fishnet? (For example it looks like Python 2.7.13 is distributed on Debian Stretch, not Python 2.7.15rc1).

**Comment by SirMorfield** [comment]
I'm not sure how I installed python and pip, there are also a lot of other small low cpu-usage programs running in the background that may have needed python 2.x to be installed.
In a couple days ill to reinstall the operating system completely, should have done that a long time ago.
I installed fishnet according to the readme:
`pip install --upgrade --user fishnet`

**Comment by Andrew-9** [comment]
I also have this issue

**Comment by niklasf** [comment]
So far I've been unable to reproduce this. The VM was running without issues for days.

Can you please provide the same details, so that we can look for any overlap?

* Copy of the backtrace
* Copy of the "Checking configuration" section at startup
* Any other information that might help reproducing this

**Comment by niklasf** [comment]
Oh ... actually I think I finally found the issue (69420af90e3427a61b569c5838020c7224ec9ad1). Fixed in v1.15.15.
